clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -433,17 +433,36 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
 
   // The call graph for this translation unit.
   CallGraph SYCLCG;
-  // The set of functions called by a kernel function.
-  llvm::SmallPtrSet<FunctionDecl *, 10> KernelSet;
+  // The set of functions called by a regular SYCL kernel.
+  llvm::SmallPtrSet<FunctionDecl *, 10> SyclCallGraphNodes;
+  // The set of functions called by an ESIMD kernel.
+  llvm::SmallPtrSet<FunctionDecl *, 10> EsimdCallGraphNodes;
   // The set of recursive functions identified while building the
   // kernel set, this is used for error diagnostics.
   llvm::SmallPtrSet<FunctionDecl *, 10> RecursiveSet;
+
+  // This is an interface function for collecting sets of
+  // functions in a SYCL call graph.
+  void collectSyclCallGraphNodes(FunctionDecl *RootNode) {
```

Comment:
Are the 2 functions required? Can't we just do the SyclSimdAttr check in this function?

Reply:
Thanks, fixed.